### PR TITLE
Various PROTO Improvements

### PR DESCRIPTION
**Pull request by DavidMansolino**
This PR improves/extends various PROTO (I will probably re-target develop instead of master later on)

https://www.cyberbotics.com/doc/guide/samples-environments?version=feature-objects-improvments#hall-wbt
https://www.cyberbotics.com/doc/guide/appearances?version=feature-objects-improvments
https://www.cyberbotics.com/doc/guide/object-apartment-structure?version=feature-objects-improvments#radiator
https://www.cyberbotics.com/doc/guide/object-factory?version=feature-objects-improvments#fire-extinguisher
https://www.cyberbotics.com/doc/guide/object-factory?version=feature-objects-improvments#capscrew
https://www.cyberbotics.com/doc/guide/object-lights?version=feature-objects-improvments#doublefluorescentlamp
https://www.cyberbotics.com/doc/guide/object-telephone?version=feature-objects-improvments

### CODE REVIEW

**Comment by DavidMansolino** [comment]
A few pictures of the new environment and objects:

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
I think the first iteration of this world is now over (probably a second iteration will be required later but it is anyway good to merge all the changes/addition to the objects now).

:+1:

File: projects/appearances/protos/OsbWood.proto [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#VRML_SIM R2019a utf8
+# license: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+# license url: https://cyberbotics.com/webots_assets_license
+# An OSB wood steel material. The color can be overridden using the `colorOverride` field.
```
Comment (omichel) [inline projects/appearances/protos/OsbWood.proto]:
```suggestion
# An OSB wood material. The color can be overridden using the `colorOverride` field.
```

File: projects/objects/lights/protos/DoubleFluorescentLamp.proto [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#VRML_SIM R2019a utf8
+# license: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+# license url: https://cyberbotics.com/webots_assets_license
+# tags: static
+# A double fluorescent lamp.
+# It does not contain any actual Light node, it only use emissive color.
```
Comment (omichel) [inline projects/objects/lights/protos/DoubleFluorescentLamp.proto]:
```suggestion
# It does not contain any actual Light node, it only uses emissive color.
```

File: projects/objects/lights/protos/DoubleFluorescentLamp.proto [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            geometry Cylinder {
+              height 1
+              radius 0.01
+            }
+          }
+        ]
+      }
+      DEF CYLINDER_LEFT Transform {
+        translation -0.015 0.055 0
+        rotation 0.9999999999999999 0 0 4.712388966
```
Comment (omichel) [inline projects/objects/lights/protos/DoubleFluorescentLamp.proto]:
```suggestion
        rotation 1 0 0 4.712388966
```

File: projects/objects/lights/protos/DoubleFluorescentLamp.proto [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        children [
+          USE LIGHT_SHAPE
+        ]
+      }
+      Transform {
+        translation -0.015 0.055 0.4999999991369187
+        rotation 0.9999999999999999 0 0 4.712388966
```
Comment (omichel) [inline projects/objects/lights/protos/DoubleFluorescentLamp.proto]:
```suggestion
        rotation 1 0 0 4.712388966
```

File: projects/objects/lights/protos/DoubleFluorescentLamp.proto [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            geometry Cylinder {
+              height 0.012
+              radius 0.012
+            }
+          }
+        ]
+      }
+      Transform {
+        translation 0.014999999999999986 0.055 0.4999999991369187
+        rotation 0.9999999999999999 0 0 4.712388966
```
Comment (omichel) [inline projects/objects/lights/protos/DoubleFluorescentLamp.proto]:
```suggestion
        rotation 1 0 0 4.712388966
```

File: projects/objects/lights/protos/DoubleFluorescentLamp.proto [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        children [
+          USE CONNECTOR
+        ]
+      }
+      Transform {
+        translation 0.014999999999999986 0.055 -0.5
+        rotation 0.9999999999999999 0 0 4.712388966
```
Comment (omichel) [inline projects/objects/lights/protos/DoubleFluorescentLamp.proto]:
```suggestion
        rotation 1 0 0 4.712388966
```

File: projects/objects/lights/protos/DoubleFluorescentLamp.proto [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+      }
+      Transform {
+        translation 0.014999999999999986 0.055 0.4999999991369187
+        rotation 0.9999999999999999 0 0 4.712388966
+        children [
+          USE CONNECTOR
+        ]
+      }
+      Transform {
+        translation 0.014999999999999986 0.055 -0.5
```
Comment (omichel) [inline projects/objects/lights/protos/DoubleFluorescentLamp.proto]:
```suggestion
        translation 0.015 0.055 -0.5
```

**Comment by DavidMansolino** [comment]
> It seems that many translation and rotation values are not rounded.

Yes indeed, that's a consequence of the fact that from Webots we save with the maximum precision. But I am not sure that we want to round them in the PROTO as it will imply thousands of manual changes each time we switch between the world file and the PROTO file.

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
👍

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
👍

File: docs/guide/object-lights.md [hunk 8]
```diff
@@ -54,6 +54,37 @@ CeilingLight {
 
 - `pointLightCastShadows`: Defines whether the point light should cast shadows.
 
+## DoubleFluorescentLamp
+
+A double fluorescent lamp.
+It does not contain any actual Light node, it only use emissive color.
```
Comment (omichel) [inline docs/guide/object-lights.md]:
```suggestion
It does not contain any actual Light node, it only uses emissive color.
```

File: projects/objects/factory/fire_extinguisher/protos/FireExtinguisher.proto [hunk 9]
```diff
@@ -0,0 +1,188 @@
+#VRML_SIM R2019a utf8
+# license: Attribution-NonCommercial 4.0 International (original model by 3DHaupt)
+# license url: https://creativecommons.org/licenses/by-nc/4.0
+# tags: static
+# A simple fire extenguisher with optional physics.
```
Comment (omichel) [inline projects/objects/factory/fire_extinguisher/protos/FireExtinguisher.proto]:
```suggestion
# A simple fire extinguisher with optional physics.
```

File: docs/guide/object-factory.md [hunk 10]
```diff
@@ -137,6 +137,38 @@ ConveyorBelt {
 
 - `speed`: Defines the rubber band speed in meters per second.
 
+## Fire Extinguisher
+
+### FireExtinguisher
+
+A simple fire extenguisher with optional physics.
```
Comment (omichel) [inline docs/guide/object-factory.md]:
```suggestion
A simple fire extinguisher with optional physics.
```

File: docs/guide/samples-environments.md [hunk 11]
```diff
@@ -45,6 +45,24 @@ These objects are Webots PROTO nodes, and can be easily moved and resized to cre
...
+%figure "Screenshots of the hall environment."
+
+|                                          |                                          |
+|------------------------------------------|------------------------------------------|
+| ![hall_a.png](images/samples/hall_a.png) | ![hall_b.png](images/samples/hall_b.png) |
+| ![hall_c.png](images/samples/hall_c.png) | ![hall_d.png](images/samples/hall_d.png) |
+
+%end
+
+This example uses some of the industrial objects included in Webots to create a training hall.
```
Comment (omichel) [inline docs/guide/samples-environments.md]:
```suggestion
This example uses some of the industrial objects included in Webots to create a factory hall.
```

File: docs/guide/samples-environments.md [hunk 12]
```diff
@@ -45,6 +45,24 @@ These objects are Webots PROTO nodes, and can be easily moved and resized to cre
...
+These objects are Webots PROTO nodes, and can be easily moved and resized to create other layouts.
```
Comment (omichel) [inline docs/guide/samples-environments.md]:
```suggestion
These objects are Webots PROTO nodes, and can be easily moved and resized to create other configurations.
```

File: projects/samples/environments/factory/worlds/hall.wbt [hunk 13]
```diff
@@ -0,0 +1,3894 @@
+#VRML_SIM R2019a utf8
+WorldInfo {
```
Comment (omichel) [inline projects/samples/environments/factory/worlds/hall.wbt]: We should add at least a title here, and maybe some info strings.

File: projects/samples/environments/factory/worlds/hall.wbt [hunk 14]
```diff
@@ -0,0 +1,3894 @@
...
+        size 1 0.1 1
+      }
+    }
+  ]
+  name "solid(9)"
+  boundingObject USE MANHOLE_SHAPE
+}
+Solid {
+  translation 8.059999958284392 -0.048 1.450000115940602
+  rotation 0 0.9999999999999999 0 4.712388966
```
Comment (omichel) [inline projects/samples/environments/factory/worlds/hall.wbt]: There are many strange values here (see rotation and translation fields).

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
👍

**Pull request merged**